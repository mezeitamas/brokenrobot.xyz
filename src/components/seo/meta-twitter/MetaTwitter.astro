---
import { SITE_METADATA } from '../../../consts';

type Props = {
    title: string;
    description: string;
    image: string;
};

const { title, description, image } = Astro.props;
---

<>
    <meta
        name="twitter:card"
        content="summary_large_image"
    />
    <meta
        name="twitter:title"
        content={title}
    />
    <meta
        name="twitter:description"
        content={description}
    />
    <meta
        name="twitter:image"
        content={new URL(image, Astro.url)}
    />
    <meta
        name="twitter:creator"
        content={SITE_METADATA.AUTHOR.SOCIAL.TWITTER_USERNAME}
    />
    <meta
        name="twitter:site"
        content={SITE_METADATA.AUTHOR.SOCIAL.TWITTER_USERNAME}
    />
</>
