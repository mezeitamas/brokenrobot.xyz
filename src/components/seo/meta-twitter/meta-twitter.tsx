import React from 'react';
import type { FunctionComponent } from 'react';

import type { Author } from '../../site-metadata/site-metadata.types';

type MetaTwitterProps = {
    title: string;
    description: string;
    url: string;
    author: Author;
};

const MetaTwitter: FunctionComponent<MetaTwitterProps> = ({ title, description, url, author }) => {
    return (
        <>
            <meta
                name="twitter:card"
                content="summary"
            />
            <meta
                name="twitter:title"
                content={title}
            />
            <meta
                name="twitter:description"
                content={description}
            />
            <meta
                name="twitter:url"
                content={url}
            />
            <meta
                name="twitter:creator"
                content={author.social.twitterUsername}
            />
        </>
    );
};

export { MetaTwitter };

export type { MetaTwitterProps };
