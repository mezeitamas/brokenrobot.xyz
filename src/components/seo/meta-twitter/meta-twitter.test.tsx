import React from 'react';

import { create } from 'react-test-renderer';

import { MetaTwitter } from './meta-twitter';
import type { MetaTwitterProps } from './meta-twitter';

describe('<MetaTwitter />', () => {
    test('should render', () => {
        // Given
        const props: MetaTwitterProps = {
            title: 'Hosting a static website on Amazon S3',
            description:
                'Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily itâ€™s quite straightforward. One option...',
            url: 'https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/',
            author: {
                name: 'Tamas Mezei',
                social: {
                    githubUrl: 'https://github.com/mezeitamas',
                    linkedinUrl: 'https://www.linkedin.com/in/mezeitamas/',
                    mastodonUrl: 'https://mastodon.social/@brokenrobot_xyz',
                    mastodonUsername: '@brokenrobot_xyz@mastodon.social',
                    twitterUrl: 'https://twitter.com/brokenrobot_xyz',
                    twitterUsername: '@brokenrobot_xyz'
                }
            }
        };

        // When
        const component = create(<MetaTwitter {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
