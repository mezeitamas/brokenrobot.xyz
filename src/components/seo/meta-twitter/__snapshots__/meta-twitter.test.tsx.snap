// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaTwitter /> should render 1`] = `
[
  <meta
    content="summary"
    name="twitter:card"
  />,
  <meta
    content="Hosting a static website on Amazon S3"
    name="twitter:title"
  />,
  <meta
    content="Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily itâ€™s quite straightforward. One option..."
    name="twitter:description"
  />,
  <meta
    content="https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/"
    name="twitter:url"
  />,
]
`;
